src/main/java/emu/grasscutter/data/excels/GadgetInteractData.java

```diff
@@ -0,0 +1,42 @@
+package emu.grasscutter.data.excels;
+
+import emu.grasscutter.data.GameResource;
+import emu.grasscutter.data.ResourceType;
+import lombok.Data;
+import lombok.Getter;
+
+@ResourceType(name = "GadgetInteractExcelConfigData.json")
+@Getter
+public class GadgetInteractData extends GameResource {
+    private int interactId;
+    private String actionType;
+    private int param1;
+    private GadgetInteractActionList[] actionList;
+    private GadgetInteractCostItems[] costItems;
+    private long uiTitleTextMapHash;
+    private long uiDescTextMapHash;
+    private GadgetInteractCondList[] condList;
+    private Boolean consumeItemNum;
+
+    @Data
+    public static class GadgetInteractActionList {
+        private String actionType;
+        private int[] param;
+    }
+
+    @Data
+    public static class GadgetInteractCostItems {
+        private int id;
+        private int count;
+    }
+
+    @Data
+    public static class GadgetInteractCondList {
+        private String condType;
```

Comment:
This could probably be moved to an enum like this:
```java
public enum InteractCondType {
    INTERACT_COND_NONE,
    INTERACT_COND_WIDGET_ON,
    INTERACT_COND_HAS_ITEM,
    INTERACT_COND_REGIONAL_PLAY_VAR_GREATER_THAN,
    INTERACT_COND_OFFERING_LEVEL
} 
```

This class is also not really a list, so it might be better to just call it `GadgetInteractCond`